source/simulator/stokes_matrix_free.cc

```diff
@@ -1722,11 +1722,16 @@ namespace aspect
                                                                                 &(sim.dof_handler));
 
                   fe_values.reinit(simulator_cell);
-                  in.reinit(fe_values, simulator_cell, sim.introspection, sim.current_linearization_point);
+                  in.reinit(fe_values, simulator_cell, sim.introspection, sim.current_linearization_point,
+                            true /* = compute_strain_rate */);
 
                   sim.material_model->fill_additional_material_model_inputs(in, sim.current_linearization_point, fe_values, sim.introspection);
                   sim.material_model->evaluate(in, out);
 
+                  Assert(std::isfinite(in.strain_rate[0].norm()),
+                         ExcMessage("Invalid strain_rate in the MaterialModelInputs. This is likely because it was "
+                                    "not filled by the caller."));
+
```

Comment:
Yeah, this might be overly defensive (I just added the check everywhere). One reason for keeping it would be that we will deprecate and replace the calls to ``reinit`` at some point soon and then this check will make sure we do it correctly.
Do you want me to remove it?